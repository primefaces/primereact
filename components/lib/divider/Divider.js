import * as React from 'react';
import { classNames } from '../utils/Utils';
import { DividerBase } from './DividerBase';

export const Divider = React.forwardRef((inProps, ref) => {
    const props = DividerBase.getProps(inProps);

    const elementRef = React.useRef(null);
    const horizontal = props.layout === 'horizontal';
    const vertical = props.layout === 'vertical';
    const otherProps = DividerBase.getOtherProps(props);
    const className = classNames(
        `p-divider p-component p-divider-${props.layout} p-divider-${props.type}`,
        {
            'p-divider-left': horizontal && (!props.align || props.align === 'left'),
            'p-divider-right': horizontal && props.align === 'right',
            'p-divider-center': (horizontal && props.align === 'center') || (vertical && (!props.align || props.align === 'center')),
            'p-divider-top': vertical && props.align === 'top',
            'p-divider-bottom': vertical && props.align === 'bottom'
        },
        props.className
    );

    React.useImperativeHandle(ref, () => ({
        props,
        getElement: () => elementRef.current
    }));

    return (
        <div ref={elementRef} className={className} style={props.style} role="separator" {...otherProps}>
            <div className="p-divider-content">{props.children}</div>
        </div>
    );
});

Divider.displayName = 'Divider';
