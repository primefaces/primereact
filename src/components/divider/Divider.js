import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { classNames } from '../utils/Utils';

export class Divider extends Component {

    static defaultProps = {
        align: null,
        layout: 'horizontal',
        type: 'solid',
        style: null,
        className: null
    }

    static propTypes = {
        align: PropTypes.string,
        layout: PropTypes.string,
        type: PropTypes.string,
        style: PropTypes.object,
        className: PropTypes.string
    };

    get isHorizontal() {
        return this.props.layout === 'horizontal';
    }

    get isVertical() {
        return this.props.layout === 'vertical';
    }

    render() {
        const dividerClassName = classNames(`p-divider p-component p-divider-${this.props.layout} p-divider-${this.props.type}`, {
            'p-divider-left': this.isHorizontal && (!this.props.align || this.props.align === 'left'),
            'p-divider-right': this.isHorizontal && this.props.align === 'right',
            'p-divider-center': (this.isHorizontal && this.props.align === 'center') || (this.isVertical && (!this.props.align || this.props.align === 'center')),
            'p-divider-top': this.isVertical && this.props.align === 'top',
            'p-divider-bottom': this.isVertical && this.props.align === 'bottom',
        }, this.props.className);

        return (
            <div className={dividerClassName} style={this.props.style} role="separator">
                <div className="p-divider-content">
                    {this.props.children}
                </div>
            </div>

        );
    }
}
