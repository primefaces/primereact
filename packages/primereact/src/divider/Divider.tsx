'use client';
import { Component } from '@primereact/core/component';
import { useDivider } from '@primereact/headless/divider';
import { styles } from '@primereact/styles/divider';
import { mergeProps } from '@primeuix/utils';
import { withComponent } from 'primereact/base';
import * as React from 'react';
import { DividerProvider } from './Divider.context';
import { defaultProps } from './Divider.props';
import { DividerContent } from './content';

export const Divider = withComponent({
    name: 'Divider',
    defaultProps,
    styles,
    setup(instance) {
        const divider = useDivider(instance.inProps);

        return divider;
    },
    render(instance) {
        const { id, props, ptmi, cx, sx } = instance;

        const rootProps = mergeProps(
            {
                id,
                className: cx('root'),
                style: sx('root'),
                role: 'separator',
                'aria-orientation': props.layout
            },
            ptmi('root')
        );

        return (
            <DividerProvider value={instance}>
                <Component instance={instance} attrs={rootProps} children={props.children} />
            </DividerProvider>
        );
    },
    components: {
        Content: DividerContent
    }
});
