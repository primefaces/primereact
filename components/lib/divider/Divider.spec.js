import '@testing-library/jest-dom';
import { Divider } from './Divider';

import { snapshot } from '../../test';

describe('Divider', () => {
    snapshot(<Divider />, 'default');
    snapshot(<Divider layout="horizontal" align="left" />, 'horizontal left');
    snapshot(<Divider layout="horizontal" align="right" />, 'horizontal right');
    snapshot(<Divider layout="horizontal" align="center" />, 'horizontal center');
    snapshot(<Divider layout="vertical" align="center" />, 'vertical center');
    snapshot(<Divider layout="vertical" align="top" />, 'vertical top');
    snapshot(<Divider layout="vertical" align="bottom" />, 'vertical bottom');
    snapshot(<Divider type="dashed" />, 'type dashed');
    snapshot(<Divider type="dotted" />, 'type dotted');
});
