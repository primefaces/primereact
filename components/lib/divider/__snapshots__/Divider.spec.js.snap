// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider default 1`] = `
<div>
  <div
    class="p-divider p-component p-divider-horizontal p-divider-solid p-divider-left"
    role="separator"
  >
    <div
      class="p-divider-content"
    />
  </div>
</div>
`;

exports[`Divider horizontal center 1`] = `
<div>
  <div
    class="p-divider p-component p-divider-horizontal p-divider-solid p-divider-center"
    role="separator"
  >
    <div
      class="p-divider-content"
    />
  </div>
</div>
`;

exports[`Divider horizontal left 1`] = `
<div>
  <div
    class="p-divider p-component p-divider-horizontal p-divider-solid p-divider-left"
    role="separator"
  >
    <div
      class="p-divider-content"
    />
  </div>
</div>
`;

exports[`Divider horizontal right 1`] = `
<div>
  <div
    class="p-divider p-component p-divider-horizontal p-divider-solid p-divider-right"
    role="separator"
  >
    <div
      class="p-divider-content"
    />
  </div>
</div>
`;

exports[`Divider type dashed 1`] = `
<div>
  <div
    class="p-divider p-component p-divider-horizontal p-divider-dashed p-divider-left"
    role="separator"
  >
    <div
      class="p-divider-content"
    />
  </div>
</div>
`;

exports[`Divider type dotted 1`] = `
<div>
  <div
    class="p-divider p-component p-divider-horizontal p-divider-dotted p-divider-left"
    role="separator"
  >
    <div
      class="p-divider-content"
    />
  </div>
</div>
`;

exports[`Divider vertical bottom 1`] = `
<div>
  <div
    class="p-divider p-component p-divider-vertical p-divider-solid p-divider-bottom"
    role="separator"
  >
    <div
      class="p-divider-content"
    />
  </div>
</div>
`;

exports[`Divider vertical center 1`] = `
<div>
  <div
    class="p-divider p-component p-divider-vertical p-divider-solid p-divider-center"
    role="separator"
  >
    <div
      class="p-divider-content"
    />
  </div>
</div>
`;

exports[`Divider vertical top 1`] = `
<div>
  <div
    class="p-divider p-component p-divider-vertical p-divider-solid p-divider-top"
    role="separator"
  >
    <div
      class="p-divider-content"
    />
  </div>
</div>
`;
